x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/conditional/CaseProcessor.java

```diff
@@ -40,11 +39,13 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public Object process(Object input) {
-        List<Object> objects = new ArrayList<>(processors.size());
-        for (Processor processor : processors) {
-            objects.add(processor.process(input));
+        for (int i = 0; i < processors.size() - 2; i += 2) {
```

Comment:
So, here you basically take a condition and you evaluate it, and if it holds then you evaluate its results and break out of the loop. If it doesn't hold (evaluates to `false`), then move on to the next set of [condition, result] and repeat the process. Can you add a comment about this, please?

Reply:
Added.